plugins/home/src/components/VisitListener.tsx

```diff
@@ -0,0 +1,163 @@
+ *   // returns "component:default/playback-order"
+ */
+export const getToEntityRef =
+  ({
+    rootPath = 'catalog',
+    stringifyEntityRefImpl = stringifyEntityRef,
+  } = {}) =>
+  ({ pathname }: { pathname: string }): string | undefined => {
+    const regex = new RegExp(
+      `^\/${rootPath}\/(?<namespace>[^\/]+)\/(?<kind>[^\/]+)\/(?<name>[^\/]+)`,
+    );
+    const result = regex.exec(pathname);
+    if (!result || !result?.groups) return undefined;
+    const entity = {
+      namespace: result.groups.namespace,
+      kind: result.groups.kind,
+      name: result.groups.name,
+    };
+    return stringifyEntityRefImpl(entity);
+  };
+
+/**
+ * @public
+ * This function returns an implementation of visitName which is responsible
+ * for receiving a pathname and returning a string (name). The default
+ * implementation ignores the pathname and uses the document.title .
+ */
+export const getVisitName = (document: Document) => () => document.title;
+
+/**
+ * @public
+ * Component responsible for listening to location changes and calling
+ * the visitsApi to save visits.
+ */
+export const VisitListener = ({
+  children,
+  toEntityRef,
+  visitName,
+}: {
+  children?: React.ReactNode;
+  toEntityRef?: ({ pathname }: { pathname: string }) => string | undefined;
+  visitName?: ({ pathname }: { pathname: string }) => string;
+}): JSX.Element => {
+  const [doNotTrack, setDoNotTrack] = useState<boolean>(
+    defaultVisitListenerContext.doNotTrack,
+  );
+  const visitsApi = useApi(visitsApiRef);
+  const { pathname } = useLocation();
+  const toEntityRefImpl = toEntityRef ?? getToEntityRef();
+  const visitNameImpl = visitName ?? getVisitName(document);
+  useEffect(() => {
+    // Wait for the browser to finish with paint with the assumption react
+    // has finished with dom reconciliation and the doNotTrack state update.
+    const requestId = requestAnimationFrame(() => {
+      if (!doNotTrack)
+        visitsApi.saveVisit({
+          visit: {
+            name: visitNameImpl({ pathname }),
+            pathname,
+            entityRef: toEntityRefImpl({ pathname }),
+          },
+        });
+    });
+    return () => cancelAnimationFrame(requestId);
+  }, [doNotTrack, visitsApi, pathname, toEntityRefImpl, visitNameImpl]);
+
+  return (
+    <VisitListenerContext.Provider value={{ doNotTrack, setDoNotTrack }}>
+      {children}
+    </VisitListenerContext.Provider>
+  );
+};
+
+/**
+ * @public
+ * Hook used to access visit listener context. Is able to control if tracking
+ * should be disabled.
+ */
+export const useVisitListener = () => {
+  const value = useContext(VisitListenerContext);
+
+  if (value === undefined)
+    throw new Error(
+      'useVisitListener found an undefined context, <VisitListener/> could be missing',
+    );
+
+  return value;
+};
+
+/**
+ * @public
+ * Use this component to warn VisitListener to disable tracking.
+ */
+export const DoNotTrack = ({
```

Comment:
This seems a bit tricky as a pattern, is it intended for use in other plugins? What's the current need for this?

Reply:
Given the `<VisitListener/>` tracks every page the user visits through the visitsAPI, there are some cases where you would not like this behaviour. One example is error pages.
There is already one alternative to this usage. Through the `useVisitListener` hook, by setting `setDoNotTrack(true)`.
In the spirit of brainstorming ideas, another path could be providing an optional filter function to `<VisitListener/>`. Therefore one could provide an implementation that could exclude the desired pages.

Reply:
wdyt?

Reply:
@Rugvip I had a chat with @tikabom and what do you think about removing any do not track functionality for the moment and discuss it in a later PR. We might implement simple allowList and denyList properties to the VisitListener component, one example proposal.

Reply:
Yep that makes sense! 👍

Reply:
Removed. I will raise another PR for the feature branch with more detailed information so we can bring this discussion there.
1a8eacdd1ee5a11f555f2824b25f41b05f9df15e